@xen-orchestra/lite/src/views/new-vm/NewVmView.vue

```diff
@@ -0,0 +1,968 @@
+  const newVdis = vmCreationParams.value.vdis
+  const existingDisks = vmCreationParams.value.existingDisks
+
+  const $defer = []
+
+  if (!templateRef) {
+    console.error('Error : templateRef is undefined or invalid.')
+    return
+  }
+
+  isBusy.value = true
+
+  try {
+    const vmRefs = vmCreationParams.value.clone
+      ? await xapi.vm.clone({ [templateRef]: newVmName })
+      : await xapi.vm.copy({ [templateRef]: newVmName }, '' as XenApiSr['$ref'])
+
+    $defer.push(() => xapi.vm.delete(vmRefs))
+
+    // Removes disks from the provision XML, we will create them by ourselves.
+    await xapi.vm.removeFromOtherConfig(vmRefs, 'disks')
+
+    // Inspects the disk configuration contained within the VM's other_config,
+    // creates VDIs and VBDs and then executes any applicable post-install script.
+    await xapi.vm.provision(vmRefs)
+
+    // INSTALL SETTINGS
+    const vm = await xapi.getField<XenApiVm>('VM', vmRefs[0], 'record')
+
+    const installMethod = vmState.selectedVdi ? 'cd' : 'network'
+
+    // TODO maybe improve this part
+    if (vm.domain_type === 'hvm') {
+      if ((newVdis.length === 0 && existingDisks.length === 0) || installMethod === 'network') {
+        const { order } = vm.HVM_boot_params
+        await xapi.call('VM.set_HVM_boot_params', [vmRefs[0], { order: order ? 'n' + order.replace('n', '') : 'ncd' }])
+      }
+    } else {
+      if (vm.PV_bootloader === 'eliloader') {
+        if (installMethod === 'network') {
+          await xapi.call('VM.set_other_config', [vmRefs[0], 'install-repository', undefined])
+        } else if (installMethod === 'cd') {
+          await xapi.call('VM.set_other_config', [vmRefs[0], 'install-repository', 'cdrom'])
+        }
+      }
+    }
+
+    if (installMethod === 'cd') {
+      const vbds = await xapi.getField<XenApiVbd['$ref'][]>('VM', vmRefs[0], 'VBDs')
+
+      for (const vbdRef of vbds) {
+        const type = await xapi.getField<string>('VBD', vbdRef, 'type')
+        if (type === 'CD') {
+          await xapi.vbd.insert(vbdRef, selectedVdiRef)
+          await xapi.vbd.setBootable(vbdRef, true)
+          return
+        }
+      }
+
+      await xapi.vbd.create({
+        vmRefs,
+        vdiRef: selectedVdiRef,
+        type: 'CD',
+        bootable: true,
+      })
+    }
+
+    // COPY BIOS strings
+    if (
+      vmState.new_vm_template?.bios_strings.length &&
+      vmCreationParams.value.hvmBootFirmware !== 'uefi' &&
+      vm.domain_type === 'hvm' &&
+      vmCreationParams.value.copyHostBiosStrings
+    ) {
+      await xapi.call('VM.copy_bios_strings', [vmRefs, vmState.new_vm_template.affinity ?? hostMasterRef])
+    }
+
+    // VIFs CREATION
+    const newVifs = vmCreationParams.value.vifs
+
+    // Direct call to the API here because otherwise we do not yet have the vmRefs of the clone or the copy before assigning the devices
+    const existingVifs = await xapi.getField<string[]>('VM', vmRefs[0], 'VIFs')
```

Comment:
Normally, your `vm` variable already have his VIFs (since you get his record directly from the XAPI and not from the cache)
```suggestion
    const existingVifs = vm.VIFs
```